tests/test_pipelines.py

```diff
@@ -0,0 +1,68 @@
+# This file is part of ap_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import glob
+import os.path
+import unittest
+
+# need to import pyproj to prevent file handle leakage
+import pyproj  # noqa: F401
+
+import lsst.pipe.base
+import lsst.utils
+import lsst.utils.tests
+
+
+class PipelineDefintionsTestSuite(lsst.utils.tests.TestCase):
+    """Tests of the self-consistency of our pipeline definitions.
+    """
+    def setUp(self):
+        self.path = os.path.join(lsst.utils.getPackageDir("ap_pipe"), "pipelines")
+
+    def test_pipelines(self):
+        """Test that each pipeline definition file in `_ingredients/` can be
+        used to build a graph.
+        """
+        files = glob.glob(os.path.join(self.path, "_ingredients/*.yaml"))
+        for file in files:
+            if "ApTemplate" in file:
+                # Our ApTemplate definition cannot be tested here because it
+                # depends on drp_tasks, which we cannot make a dependency here.
+                continue
+            with self.subTest(file):
+                pipeline = lsst.pipe.base.Pipeline.from_uri(file)
+                if "apPipe" in pipeline.subsets:
+                    pipeline.addConfigPython("diaPipe", "config.apdb.db_url='sqlite:tests/data/test_apdb.db'")
```

Comment:
Why not:
```suggestion
                    pipeline.addConfigPython("diaPipe", "config.apdb.db_url='sqlite://'")
```
It's valid, and guarantees no side effects.

Reply:
I’m confused on why this needs to be a python config block rather than just a regular config override

Reply:
I didn't know that was an option: I see it now as `addConfigOverride`, which is definitely clearer.